zerver/lib/actions.py

```diff
@@ -3427,3 +3429,56 @@ def do_set_name_changes_disabled(realm, name_changes_disabled):
         value=name_changes_disabled,
     )
     send_event(event, active_user_ids(realm))
+
+def notify_realm_custom_profile_fields(realm):
+    # type: (Realm) -> None
+    fields = custom_profile_fields_for_realm(realm.id)
+    event = dict(type="custom_profile_fields", fields=fields)
+    send_event(event, active_user_ids(realm))
+
+def do_add_realm_custom_profile_field(realm, name, field_type):
+    # type: (Realm, Text, Text) -> int
+    with transaction.atomic():
+        custom_field = CustomProfileField(realm=realm, name=name,
+                                          field_type=field_type)
+        custom_field.full_clean()
+        custom_field.save()
+        realm.profile_fields.append(custom_field.as_dict())
+        realm.save(update_fields=['profile_fields'])
+
+    notify_realm_custom_profile_fields(realm)
+    return custom_field.pk
```

Comment:
Use `.id`, not `.pk`; we as a convention just use the one in the codebase to avoid confusing folks new to Django with 2 different names for the same thing.